lib/src/util/dart_type_utilities.dart

```diff
@@ -13,9 +13,14 @@ typedef bool AstNodePredicate(AstNode node);
 
 class DartTypeUtilities {
   static bool extendsClass(DartType type, String className, String library) =>
-      isClass(type, className, library) ||
-      (type is InterfaceType &&
-          extendsClass(type.superclass, className, library));
+      _extendsClass(type, new Set<ClassElement>(), className, library);
+
+  static bool _extendsClass(DartType type, Set<ClassElement> seenTypes,
+          String className, String library) =>
+      type is InterfaceType &&
+      seenTypes.add(type.element) &&
+      (isClass(type, className, library) ||
+          (_extendsClass(type.superclass, seenTypes, className, library)));
```

Comment:
nit: You could remove the parens around the invocation of `_extendsClass`.

Reply:
👍 Thanks!